java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -152,7 +152,16 @@ public Block entry() {
     return blocks;
   }
 
-  public static class Block {
+  public interface IBlock<T> {
+    int id();
+    List<T> elements();
+
+    T terminator();
+
+    Set<? extends IBlock<T>> successors();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1452)